### PR TITLE
Add support for /compose/metadata, /compose/results, as well as tests

### LINKED ISSUE #643: tests: composer-cli compose metadata fails
```
# composer-cli compose list
0396b0be-410b-44b6-97dd-e317f2b17c52 FINISHED test-blueprint 0.0.2 AWS

# composer-cli --json compose metadata 0396b0be-410b-44b6-97dd-e317f2b17c52
No Content-Disposition header; cannot get filename
```
exit code 1.

This causes the newly enabled sanity tests in #642 to fail.

### PR DESCRIPTION
**Pull request by bcl**
This fixes #643 and #644

### CODE REVIEW

**Comment by bcl** [comment]
Rebased on master, new vmdk failure seems unrelated.

File: internal/weldr/api.go [hunk 1]
```diff
@@ -2025,6 +2026,67 @@ func (api *API) composeImageHandler(writer http.ResponseWriter, request *http.Re
...
+	if composeStatus.State != common.CFinished && composeStatus.State != common.CFailed {
+		errors := responseError{
+			ID:  "BuildInWrongState",
+			Msg: fmt.Sprintf("Build %s is in wrong state: %s", uuidString, composeStatus.State.ToString()),
+		}
+		statusResponseError(writer, http.StatusBadRequest, errors)
+		return
+	}
+
+	metadata, err := json.Marshal(compose)
```
Comment (teg) [inline internal/weldr/api.go:2067]: I think we should just expose the manifest here (the `Compose` is really internal state). So I would use `compose.ImageBuild.Manifest` here instead.

File: internal/worker/server.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func (s *Server) JobStatus(id uuid.UUID) (*JobStatus, error) {
 	if canceled {
 		state = common.CFailed
 	} else if !finished.IsZero() {
-		if result.OSBuildOutput.Success {
+		if result.OSBuildOutput != nil && result.OSBuildOutput.Success {
```
Comment (teg) [inline internal/worker/server.go:100]: Hm, as discussed on the other PR, this seems to be masking some underlying problem. I'd have loved to figure out the underlying issue there was. Is this a blocker for the rest of this PR? (My concern would be that even if this avoids us from crashing, we are still in an undefined/inconsistent state that we should probably have caught/handled elsewhere).

File: internal/weldr/api.go [hunk 3]
```diff
@@ -2087,6 +2088,98 @@ func (api *API) composeMetadataHandler(writer http.ResponseWriter, request *http
...
```
Comment (teg) [inline internal/weldr/api.go:2128]: Same comment as on the other patch.

File: internal/worker/server.go (hunk 2 above)
Comment (bcl) [reply internal/worker/server.go:100]: Depending on the state of the system it fails. On a clean install I think it is fine -- but in my opinion if there's a pointer involved there is always going to be the potential for it to be nil and that should be checked, or the pointer should be eliminated.
I can drop this from this PR and we can cover it in the other one, I added it here so I could run the tests in my VM.

File: internal/weldr/api.go (hunk 1 above)
Comment (bcl) [reply internal/weldr/api.go]: Ok, added that, but also a check for nil that returns a 400 with an 'EmptyManifest' error.

File: internal/worker/server.go (hunk 2 above)
Comment (teg) [reply internal/worker/server.go]: If it doesn't block this, then let's continue the discussion on the other PR

**Comment by teg** [comment]
On closer inspection, the only tests that fail are the RCM tests, which were known to be broken and have been fixed on master. They are also going to be dropped and are independent of this PR. Merging.

**Pull request merged**